lib/data_integrity_utils.pm

```diff
@@ -19,22 +19,43 @@ use testapi;
 use File::Basename;
 use Digest::file 'digest_file_hex';
 
-our @EXPORT = 'verify_checksum';
+our @EXPORT = qw(verify_checksum get_image_digest);
+
+=head2
+Returns image digest. Digest retrieval is platform-specific depends.
+=cut
+sub get_image_digest {
+    my ($image_path) = shift;
+
+    my $digest;
+    if (check_var('BACKEND', 'svirt')) {
+        $digest = console('svirt')->get_cmd_output("sha256sum $image_path");
+        # On Hyper-V the hash starts with '\'
+        my $start = check_var('VIRSH_VMM_FAMILY', 'hyperv') ? 1 : 0;
+        $digest = substr $digest, $start, 64;    # extract SHA256 from the output
+    }
+    else {
+        $digest = digest_file_hex($image_path, "SHA-256");
+    }
+    return $digest;
+}
 
 =head2 verify_checksum
 Verify image data integrity by comparing SHA256 checksums
 =cut
 sub verify_checksum {
-    my ($dir_path) = shift;    # for backends other than qemu, image directory path needs to be set
+    my ($dir_path) = shift;
 
     diag "Comparing data integrity calculated with SHA256 digest against checksum from IBS/OBS via rsync.pl";
     foreach my $image (grep { /^CHECKSUM_/ } keys %bmwqemu::vars) {
         my $checksum = get_required_var $image;
         $image =~ s/CHECKSUM_//;
         my $image_path = get_required_var $image;
+        $dir_path = '/var/lib/libvirt/images/' if check_var('BACKEND', 'svirt');
```

Comment:
I would suggest to do this two assignation from outside as the method is prepared for accept $dir_path values. Or if you don't want to use it, remove the parameter and extract similar method that you did for `get_image_digest`

Reply:
OK, I pushed them to the test itself.